app/vmctl/vm_native.go

```diff
@@ -182,9 +184,14 @@ func (p *vmNativeProcessor) runSingle(ctx context.Context, f filter, srcURL, dst
 		w = limiter.NewWriteLimiter(pw, rl)
 	}
 
-	_, err = io.Copy(w, barReader)
+	cb := func() error {
+		_, err = io.Copy(w, barReader)
+		return err
+	}
```

Comment:
Are you sure this operation can be safely retried?
What happens if read N bytes from Reader and failed to write them into the Writer? Will the subsequent call re-read those N bytes again or it will read the next M bytes?